client/coral-admin/src/containers/Streams/Streams.js

```diff
@@ -0,0 +1,182 @@
+import React, {Component} from 'react';
+import styles from './Streams.css';
+import {connect} from 'react-redux';
+import I18n from 'coral-framework/modules/i18n/i18n';
+import {fetchAssets, updateAssetState} from '../../actions/assets';
+import translations from '../../translations.json';
+import {
+  RadioGroup,
+  Radio,
+  Icon,
+  DataTable,
+  TableHeader
+} from 'react-mdl';
+import Pager from 'coral-ui/components/Pager';
+
+const limit = 25;
+
+class Streams extends Component {
+
+  state = {
+    search: '',
+    sort: 'desc',
+    filter: 'all',
+    statusMenus: {},
+    timer: null,
+    page: 0
+  }
+
+  componentDidMount () {
+    this.props.fetchAssets(0, limit, '', this.state.sortBy);
+  }
+
+  onSettingChange = (setting) => (e) => {
+    let options = this.state;
+    this.setState({[setting]: e.target.value});
+    options[setting] = e.target.value;
+    this.props.fetchAssets(0, limit, options.search, options.sort, options.filter);
+  }
+
+  onSearchChange = (e) => {
+    this.setState({search: e.target.value});
+    this.setState((prevState) => {
+      clearTimeout(prevState.timer);
+      const fetchAssets = this.props.fetchAssets;
+      prevState.timer = setTimeout(() => {
+        fetchAssets(0, limit, this.state.search, this.state.sort, this.state.filter);
+      }, 350);
+      return prevState;
+    });
+  }
+
+  renderDate = (date) => {
+    const d = new Date(date);
+    return `${d.getMonth() + 1}/${d.getDate()}/${d.getFullYear()}`;
```

Comment:
I thought about this, but it seemed out of scope for this story. It involves an unrelated setting and associated UI.